csrc/codegen.cpp

```diff
@@ -1352,11 +1365,88 @@ class CudaKernelGenerator : private kir::ConstIrVisitor {
     }
   }
 
+  void genBlockWelford(const WelfordOp* wop) {
```

Comment:
There was a `if (has_block_reduce)` guarding this, but now that check is gone. Is this the expected behavior?

Reply:
Thanks for catching this. Yes, it should be guarded.